src/btree/bt_ovfl.c

```diff
@@ -91,134 +89,56 @@ __wt_ovfl_read(WT_SESSION_IMPL *session,
-
-	*updp = NULL;
+	WT_OVFL_TRACK *track;
 
 	/* Read the overflow value. */
 	WT_RET(__wt_scr_alloc(session, 1024, &tmp));
 	WT_ERR(__wt_dsk_cell_data_ref(session, page->type, unpack, tmp));
 
-	/*
-	 * Create an update entry with no transaction ID to ensure global
-	 * visibility, append it to the update list.
-	 *
-	 * We don't need locks or barriers in this function: any thread reading
-	 * the update list will see our newly appended record or not, it doesn't
-	 * matter until the on-page cell is set to WT_CELL_VALUE_OVFL_RM. That
-	 * involves atomic operations which will act as our barrier. Regardless,
-	 * we update the page footprint as part of this operation, which acts as
-	 * a barrier as well.
-	 *
-	 * The update transaction ID choice is tricky, to work around an issue
-	 * in variable-length column store. Imagine an overflow value with an
-	 * RLE greater than 1. We append a copy to the end of an update chain,
-	 * but it's possible it's the overflow value for more than one record,
-	 * and appending it to the end of one record's update chain means a
-	 * subsequent enter of a globally visible value to one of the records
-	 * would allow the truncation of the overflow chain that leaves other
-	 * records without a value. If appending such an overflow record, set
-	 * the transaction ID to the first possible transaction ID. That ID is
-	 * old enough to be globally visible, but we can use it as a flag if an
-	 * update record cannot be discarded when truncating an update chain.
-	 */
-	WT_ERR(__wt_update_alloc(
-	    session, tmp, &append, &size, WT_UPDATE_STANDARD));
-	append->txnid = page->type == WT_PAGE_COL_VAR &&
-	    __wt_cell_rle(unpack) > 1 ? WT_TXN_FIRST : WT_TXN_NONE;
-	for (upd = upd_list; upd->next != NULL; upd = upd->next)
-		;
-	WT_PUBLISH(upd->next, append);
-
-	__wt_cache_page_inmem_incr(session, page, size);
-
-	*updp = append;
-
-err:	__wt_scr_free(session, &tmp);
-	return (ret);
-}
-
-/*
- * __ovfl_cache --
- *	Cache an overflow value.
- */
-static int
-__ovfl_cache(WT_SESSION_IMPL *session,
-    WT_PAGE *page, WT_UPDATE *upd_list, WT_CELL_UNPACK *unpack)
-{
-	WT_OVFL_TRACK *track;
-	WT_UPDATE *upd;
-
-	/* Append a copy of the overflow value to the update list. */
-	WT_RET(__ovfl_cache_append_update(
-	    session, page, upd_list, unpack, &upd));
-
 	/* Allocating tracking structures as necessary. */
 	if (page->modify->ovfl_track == NULL)
-		WT_RET(__wt_ovfl_track_init(session, page));
+		WT_ERR(__wt_ovfl_track_init(session, page));
 	track = page->modify->ovfl_track;
 
-	/* Add the value's information to the update list. */
-	WT_RET(__wt_realloc_def(session,
+	/* Copy the overflow item into place. */
+	WT_ERR(__wt_realloc_def(session,
 	    &track->remove_allocated, track->remove_next + 1, &track->remove));
 	track->remove[track->remove_next].cell = unpack->cell;
-	track->remove[track->remove_next].upd = upd;
+	WT_ERR(__wt_strndup(session,
+	    tmp->data, tmp->size, &track->remove[track->remove_next].data));
```

Comment:
To be clear, when we're duplicating strings, we should use `__wt_strndup`.  Only in the handful of cases that make copies of non-string data should we change to `__wt_memdup`.